Spec/RealtimeClientConnection.swift

```diff
@@ -2649,6 +2649,118 @@ class RealtimeClientConnection: QuickSpec {
+                                expect(client.connection.id).toNot(beNil())
+                                connectionId = client.connection.id!
+                                client.connectionStateTtl = customTtlInterval
+                                client.maxIdleInterval = customIdleInterval
+                                client.connection.once(.disconnected) { _ in
+                                    let disconnectedAt = Date()
+                                    expect(client.connectionStateTtl).to(equal(customTtlInterval))
+                                    expect(client.maxIdleInterval).to(equal(customIdleInterval))
+                                    client.connection.once(.connecting) { _ in
+                                        let reconnectionInterval = Date().timeIntervalSince(disconnectedAt)
+                                        expect(reconnectionInterval).to(beGreaterThan(client.connectionStateTtl + client.maxIdleInterval))
+                                        client.connection.once(.connected) { _ in
+                                            expect(client.connection.id).toNot(equal(connectionId))
+                                            done()
+                                        }
+                                    }
+                                }
+                                client.onDisconnected()
+                            }
+                        }
+                    }
+                    // RTN15g3
+                    it("reattaches to the same channels after a new connection has been established") {
+                        client = AblyTests.newRealtime(options)
+                        client.connect()
+                        defer { client.close() }
+                        let channelName = "test-reattach-after-ttl"
+                        let channel = client.channels.get(channelName)
+                        
+                        waitUntil(timeout: testTimeout) { done in
+                            client.connection.once(.connected) { _ in
+                                connectionId = client.connection.id!
+                                client.connectionStateTtl = customTtlInterval
+                                client.maxIdleInterval = customIdleInterval
+                                channel.attach { error in
+                                    if let error = error {
+                                        fail(error.message)
+                                    }
+                                    expect(channel.state).to(equal(ARTRealtimeChannelState.attached))
+                                }
+                                client.connection.once(.disconnected) { _ in
+                                    client.connection.once(.connecting) { _ in
+                                        client.connection.once(.connected) { _ in
+                                            expect(client.connection.id).toNot(equal(connectionId))
```

Comment:
Why is the connection ID changing?  There is no pause as far as I can see.

Reply:
This `it` block uses the same settings declared up here https://github.com/ably/ably-ios/blob/develop/Spec/RealtimeClientConnection.swift#L2655

Reply:
As above, there should be no pause according to https://docs.ably.io/client-lib-development-guide/features/#RTN15a